src/test/compile-fail/issue-32128.rs

```diff
@@ -19,7 +19,9 @@ fn main() {
         })
     };
 
-    demo.example(1);    //~ ERROR no method named `example`
-                        //~^ NOTE use `(demo.example)(...)`
+    demo.example(1);
+    //~^ ERROR no method named `example`
+    //~| HELP use `(demo.example)(...)`
+    //~| NOTE `example` is a field storing a function, not a method
```

Comment:
really all of these feel like ui tests to me :) 

maybe make a directory like `src/test/ui/suggestions/confuse-field-and-method/` and just move the (not renamed) `issue-123.rs` files into that directory?